### ISSUE TITLE
Unexpected image layout validation error in VK-GL-CTS renderpass tests

### ISSUE THREAD

**Issue by rg3igalia** [KhronosGroup/Vulkan-ValidationLayers#1700]
VK-GL-CTS has a set of tests related to render passes. Many of the tests produce validation errors related to invalid image layouts, but at least some of them appear to be unexpected. For example, dEQP-VK.renderpass2.suballocation.formats.r8_unorm.input.clear.store.clear is a test doing the following:

* Creating a render pass with two color attachments and two subpasses. The initial and final layouts for both attachments is VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL.
* The first color attachment is cleared in the first subpass.
* The first color attachment is then used as an input attachment in the second subpass, which uses the second color attachment as the output. In this second subpass, the image layout for the first color attachment is set to VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL to use it as the input attachments.
* Finally, attachments are copied to buffers to verify their contents.
* 3 command buffers are used for the operations above: images are initialized using one command buffer, the render pass is recorded using a second one, and image copying is recorded in a third one.

This test produces the following validation error:

```
Validation Error: [ UNASSIGNED-CoreValidation-DrawState-InvalidImageLayout ] Object 0: handle = 0x5ce9870, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0x4dae5635 | Submitted command buffer expects VkImage 0x5ced370[] (subresource: aspectMask 0x1 array layer 0, mip level 0) to be in layout VK_IMAGE_LAYOUT_SHADER_READ_ONLY_OPTIMAL--instead, current layout is VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL.
```

Barriers and transitions appear to be OK as far as I can see. Furthermore, dEQP-VK.renderpass.suballocation.formats.r8_unorm.input.clear.store.clear is an equivalent test using renderpass structures instead of renderpass2 structures, and produces no validation errors.

The code for both tests can be found around here:
https://github.com/KhronosGroup/VK-GL-CTS/blob/511650cdaeef498b12388ee582815e664d763bd1/external/vulkancts/modules/vulkan/renderpass/vktRenderPassTests.cpp#L4624

**Comment by Tony-LunarG** [comment]
Fixed with #1748

**Comment by rg3igalia** [comment]
Thanks!
